content/docs/hooks-overview.md

```diff
@@ -153,32 +153,32 @@ function FriendStatusWithCounter(props) {
+>आप एक समर्पित पेज पर `useEffect` के बारे में अधिक जान सकते है: [Effect Hook का उपयोग करना](/docs/hooks-effect.html)
 
-## ✌️ Rules of Hooks {#rules-of-hooks}
+## ✌️ Hooks के नियम {#rules-of-hooks}
 
-Hooks are JavaScript functions, but they impose two additional rules:
+Hooks JavaScript फंक्शन्स है, लेकिन वे दो अतिरिक्त नियम लागू करते है:
 
-* Only call Hooks **at the top level**. Don’t call Hooks inside loops, conditions, or nested functions.
-* Only call Hooks **from React function components**. Don’t call Hooks from regular JavaScript functions. (There is just one other valid place to call Hooks -- your own custom Hooks. We'll learn about them in a moment.)
+* Hooks को केवल **सबसे उपर** कॉल करें। लूप्स, कंडीशंस, या नेस्टेड फ़ंक्शन के अंदर Hooks को कॉल न करें।
+* Hooks को सिर्फ **React फ़ंक्शन कौम्पोनॅन्ट** मैं ही कॉल करें। नियमित जावास्क्रिप्ट फ़ंक्शंस से Hooks कॉल न करें। (Hooks को कॉल करने के लिए सिर्फ एक अन्य वैध स्थान है -- अपने स्वयं के कस्टम Hooks। हम उनके बारे में एक क्षण में जानेंगे।)
 
-We provide a [linter plugin](https://www.npmjs.com/package/eslint-plugin-react-hooks) to enforce these rules automatically. We understand these rules might seem limiting or confusing at first, but they are essential to making Hooks work well.
+हम एक [लिंटर प्लगइन](https://www.npmjs.com/package/eslint-plugin-react-hooks) यूज़ करेंगे जो इन नियमों को खुद लागू कर देगा। हम समझते है कि ये नियम शुरू मैं थोड़े सीमित या मुश्किल लग सकते हैं, लेकिन वे Hooks को अच्छे से काम करने के लिए आवश्यक है।
 
->Detailed Explanation
+>विस्तृत विवरण
 >
->You can learn more about these rules on a dedicated page: [Rules of Hooks](/docs/hooks-rules.html).
+>आप एक समर्पित पेज पर इन नियमों के बारे में अधिक जान सकते है: [Hooks के नियम](/docs/hooks-rules.html)
 
-## 💡 Building Your Own Hooks {#building-your-own-hooks}
+## 💡 अपने खुद के Hooks बनाना {#building-your-own-hooks}
 
-Sometimes, we want to reuse some stateful logic between components. Traditionally, there were two popular solutions to this problem: [higher-order components](/docs/higher-order-components.html) and [render props](/docs/render-props.html). Custom Hooks let you do this, but without adding more components to your tree.
+कभी-कभी, हम कौम्पोनॅन्टस के बीच कुछ स्टेटफुल लॉजिक का पुन: उपयोग करना चाहते है। परंपरागत रूप से, इस समस्या के दो लोकप्रिय समाधान थे: [higher-order कौम्पोनॅन्ट](/docs/higher-order-components.html) और [रेंडर प्रॉप्स](/docs/render-props.html)। लेकिन कस्टम Hooks आपको आपके tree में अधिक कौम्पोनॅन्ट को ऐड करे बिना ऐसा करने देते है।
 
-Earlier on this page, we introduced a `FriendStatus` component that calls the `useState` and `useEffect` Hooks to subscribe to a friend's online status. Let's say we also want to reuse this subscription logic in another component.
+इस पेज मैं पहले, हमने एक `FriendStatus` कौम्पोनॅन्ट पेश किया है, जो किसी मित्र के ऑनलाइन स्टेटस की सदस्यता के लिए `useState` और `useEffect` Hook को कॉल करता है। मान लें कि हम इस सदस्यता तर्क को किसी अन्य कौम्पोनॅन्ट में पुन: उपयोग करना चाहते है।
```

Comment:
किया tha

Reply:
सदस्यता could be replaced by a better word

Reply:
मान लें कि हम इस सदस्यता तर्क को किसी अन्य कौम्पोनॅन्ट में पुन: उपयोग करना चाहते है। revisit